src/backend/sev/builder.rs

```diff
@@ -29,6 +29,64 @@ impl kvm::Hook for Sev {
         measure::Kind::Sha256
     }
 
+    fn shim_loaded(
+        &mut self,
+        _vm: &mut VmFd,
+        addr_space: &mut [u8],
+        shim: &Component,
+    ) -> Result<()> {
+        // The initial page tables of the shim must be corrected
+        // with the current CPU's C-bit indicating encrypted memory.
+
+        // query the C-bit location via cpuid
+        use core::arch::x86_64::__cpuid_count;
+        let c_bit_loc = unsafe { __cpuid_count(0x8000_001f, 0x0000_0000) }.ebx;
```

Comment:
Technically, this is UB.

Reply:
@npmccallum: why?

@connorkuehl: that was your code :)

Reply:
@haraldh Because issuing the CPUID instruction without determining whether or not the instruction is supported by the processor results in UB. This is a complicated problem. See my comment [here](https://github.com/enarx/sgx/blob/main/src/types/secs.rs#L66). And [here](https://github.com/rust-lang/rust/issues/60123#issuecomment-581621838).

Now, any CPU which supports SEV will also, by definition, support CPUID. But it is still potentially UB.

Reply:
So, shall we copy&paste `core::arch::x86_64::has_cpuid()` and use it in our code?

Reply:
This code is unreachable without having passed all of the SEV checks which would guarantee this cpuid is correct.